src/site/content/en/blog/app-history-api/index.md

```diff
@@ -0,0 +1,402 @@
+    // User submitted a POST form to a same-domain URL
+    // (If canTransition is false, the event is just informative:
+    // you can't intercept this request, although you could
+    // likely still call .preventDefault() to stop it completely).
+
+    const submitToServer = async () => {
+      await fetch(event.destination.url, { method: 'POST', body: event.formData });
+      // You could navigate again with {replace: true} to change the URL here
+    };
+    event.transitionWhile(submitToServer());
+  }
+});
+```
+
+## What's missing?
+
+Despite the centralized nature of the "navigate" event handler, the current App History API specification doesn't trigger "navigate" on a page's first load.
+And for sites which use [Server Side Rendering][ssr-definition] (SSR) for all states, this might be fine—your server could return the correct initial state, which is the fastest way to get content to your users.
+But sites that leverage mostly JS to create their pages may need to create an additional function to initialize their page.
+This is up for discussion [in the app-history repo][initial-event-discuss].
+
+Another intentional design choice of the App History API is that it operates only within a single frame—that is, the top-level page, or a single specific `<iframe>`.
+This has a number of interesting implications that are [further documented in the spec][backforward-note], but in practice, will reduce developer confusion.
+The previous History API has a number of confusing edge cases, like support for frames, and the reimagined App History API avoids them from the get-go.
+
+{% Aside %}
+
+In the near future, it's hoped that [an unrelated change to the HTML spec][iframe-historyless] could introduce "historyless" IFrames which do not participate in the browser's history whatsoever.
+IFrames which change their URL have classically confused both developers and users because this change has no effect on the user's URL bar or overall page title, so, e.g., a user pressing Back or Forward in their browser might not have an immediately obvious effect.
+
+{% endAside %}
+
+Lastly, there's not yet consensus on programatically modifying or rearranging the list of entries the user has navigated through while using your site.
+This is [currently under discussion][bug-edit-entries], but one option could be to allow only deletions: either historic entries or "all future entries".
+The latter would allow temporary state, e.g., as a developer, I could:
+
+- ask the user a question by navigating to new URL or state
+- allow the user to complete their work (or go Back)
+- on completion, the App History API can remove that entry
+
+This could be perfect for temporary modals or interstitals: the new URL is something that a user can use the Back gesture to leave from, but they then cannot accidentaly go Forward to open it again (because the entry has been removed), and is just not possible with the current History API.
```

Comment:
```suggestion
This could be perfect for temporary modals or interstitals: the new URL is something that a user can use the Back gesture to leave from, but they then cannot accidentaly go Forward to open it again (because the entry has been removed). This is just not possible with the current History API.
```